sql/src/main/java/io/crate/planner/projection/Projection.java

```diff
@@ -88,4 +83,8 @@ public static Projection fromStream(StreamInput in) throws IOException {
     public int hashCode() {
         return projectionType().hashCode();
     }
+
+    public Map<String, Object> mapRepresentation() {
```

Comment:
What intermediate map? I didn't add any `super.mapRepresentation()` calls (though maybe I should?) . And even with a visitor / the old pattern with a Map builder we'd have intermediate builder instances, which are as heavy as a map. 

As stated in the PR description. I moved away from the visitor to make this functionality more apparent. I'm not completely opposed to keep using the visitor, but I think avoiding map instances would be the wrong reason. I think this is mostly a matter of "how much coupling do we want".

More coupling -> it's harder to miss/forget implementing the explain representation if a projection is added
Less coupling -> Well, less coupling?